torch/distributions/negative_binomial.py

```diff
@@ -99,8 +99,7 @@ def log_prob(self, value):
         log_unnormalized_prob = (self.total_count * F.logsigmoid(-self.logits) +
                                  value * F.logsigmoid(self.logits))
 
-        log_normalization = (-torch.lgamma(self.total_count + value) + torch.lgamma(1. + value) +
-                             torch.lgamma(self.total_count))
-        log_normalization[self.total_count + value == 0.] = 0.
+        log_normalization = (-torch.lgamma(self.total_count + value).nan_to_num(nan=float("nan"), posinf=0, neginf=-float("inf")) +
+                             torch.lgamma(1. + value) + torch.lgamma(self.total_count))
```

Comment:
Could you add a short comment about this issue?

Reply:
Of course, but I am not sure what you mean... I have added a commit where I use the fix proposed by @fritzo and I added a comment in the code.

        # The case self.total_count == 0 and value == 0 has probability 1 but
        # lgamma(0) is infinite. Handle this case separately using a function
        # that does not modify tensors in place to allow Jit compilation.

There are very few comments overall, so that's perhaps not what you meant. Let me know if you meant something different. I'll remove the comment from the code and write a text to the same effect along your guidelines.